lib/prometheus-parser/src/line.rs

```diff
@@ -560,6 +560,28 @@ mod test {
         assert_eq!(left, tail);
         assert!(r.is_nan());
 
+        let input = wrap(&(u64::MAX).to_string());
+        let (left, r) = Metric::parse_value(&input).unwrap();
+        assert_eq!(left, tail);
+        assert_eq!(r as u64, u64::MAX);
+
+        // This doesn't pass because we are parsing as f64 so we lose accuracy.
+        // However, according to the Prometheus docs:
+        // https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-details
+        // > value is a float represented as required by Go's ParseFloat() function.
+        // Go's `ParseFloat` will also parse `u64::MAX` and `u64::MAX - 1` as the
+        // same number, so we are in alignment with Prometheus.
+        //
+        // let input = wrap(&(u64::MAX - 1).to_string());
+        // let (left, r) = Metric::parse_value(&input).unwrap();
+        // assert_eq!(left, tail);
+        // assert_eq!(r as u64, u64::MAX - 1);
+
+        let input = wrap(&(u32::MAX as u64 + 1).to_string());
+        let (left, r) = Metric::parse_value(&input).unwrap();
+        assert_eq!(left, tail);
+        assert_eq!(r as u64, u32::MAX as u64 + 1);
+
```

Comment:
I think you can reduce this section down to a smaller comment that explains that converting from an integer to a floating-point number will start to lose precision once the number is bigger than 2^N, where N is the number of bits in the mantissa (53 bits for an `f64`).

In addition, or alternatively, you could add an assertion that we handle 2^N-1 and 2^N without loss, but that 2^N+1 rounds down to 2^N, to show we're providing a maximally correct conversion where possible.